elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/MetaDataStore.java

```diff
@@ -100,8 +131,10 @@ public void populateEntityDictionary(EntityDictionary dictionary) {
      * @param table table metadata
      */
     public void addTable(Table table) {
-        tables.put(dictionary.getEntityClass(table.getName(), table.getVersion()), table);
-        addMetaData(table);
+        String version = table.getVersion();
+        EntityDictionary dictionary = hashMapDataStores.computeIfAbsent(version, ERROR_OUT).getDictionary();
```

Comment:
This would throw an internal error (as it doesn't relate to the client request, but rather the setup of the server).

Reply:
changed to InternalServerErrorException .